pkg/proc/native/proc_freebsd.go

```diff
@@ -269,10 +298,10 @@ func (dbp *nativeProcess) trapWaitInternal(pid int, halt bool) (*nativeThread, e
 					}
 					return nil, err
 				}
-				if halt {
-					return nil, nil
+				if mode == trapWaitStepping {
+					dbp.execPtraceFunc(func() { ptraceSuspend(tid) })
```

Comment:
Is there any reason not to do this unconditionally? By that I mean just automatically suspending the thread in `dbp.addThread` for FreeBSD.

Reply:
If this happens when we are stepping (mode == trapWaitStepping) we want to suspend new threads, like every other thread is suspended, and continue until the thread we are stepping finishes stepping.
If this happens when we are running the process (mode == trapWaitNormal) we want the new thread to keep running, like every other thread.